pandas/io/pytables.py

```diff
@@ -5032,14 +5015,12 @@ def _maybe_adjust_name(name: str, version) -> str:
     -------
     str
     """
-    try:
+    with suppress(IndexError):
```

Comment:
let's do appropriate checks for this and avoid the exception altogether

Reply:
I checked that the version number is a correct sequence of three elements.  I also added test as this check never hit.